libcudacxx/include/cuda/__memcpy_async/memcpy_async.h

```diff
@@ -22,19 +22,17 @@
 #  pragma system_header
 #endif // no system header
 
-#if _CCCL_CUDA_COMPILATION()
```

Comment:
I believe we still need `_CCCL_HAS_CUDA_COMPILER()`, just not `_CCCL_CUDA_COMPILATION()`